python/caffe/io.py

```diff
@@ -81,6 +81,39 @@ def array_to_datum(arr, label=None):
     return datum
 
 
+def pad_bbox(arr, max_bboxes=64, bbox_width=16):
+    if arr.shape[0] > max_bboxes:
+        raise ValueError(
+            'Too many bounding boxes (%d > %d)' % arr.shape[0], max_bboxes
+        )
+    # fill remainder with zeroes:
+    data = np.zeros((max_bboxes+1, bbox_width), dtype='float')
+    # number of bounding boxes:
+    data[0][0] = arr.shape[0]
+    # width of a bounding box:
+    data[0][1] = bbox_width
+    # bounding box data. Merge nothing if no bounding boxes exist.
+    if arr.shape[0] > 0:
+        data[1:1 + arr.shape[0]] = arr
+
+    return data
+
+
+def bbox_to_datum(arr, label=0, max_bboxes=64, bbox_width=16):
+    """Converts a 1-dimensional bbox array to datum.
+    """
+    arr = pad_bbox(arr, max_bboxes, bbox_width)
+
+    datum = caffe_pb2.Datum()
+    datum.channels = 1
+    datum.channels, datum.height, datum.width = 1, arr.shape[0], arr.shape[1]
+
+    datum.float_data.extend(arr.flat)
+    datum.label = label
+
+    return datum
+
+
```

Comment:
None of these changes really belong in `caffe.io`. They should be implemented within the same scripts that create the LMDBs.